### PR TITLE
[GVN-266] fix loading airdrop

### PR DESCRIPTION
**Pull request by qiqicodes**
- [x] add DataSizeFilter to filter the latest airdrop

### CODE REVIEW

File: src/rewards/airdrop.ts [hunk 1]
```diff
@@ -205,15 +205,12 @@ export class Airdrop {
...
-    const stakePoolFilter: MemcmpFilter = {
-      memcmp: {
-        offset: 8,
-        bytes: stakePoolVault.toBase58()
-      }
+  static async loadAll(rewardsProgram: Program): Promise<Airdrop[]> {
+    const dataSizeFilter: DataSizeFilter = {
+      dataSize: 400464
     }
+    const airdropInfos = await rewardsProgram.account.airdrop.all([dataSizeFilter])
```
Comment (Tristyn) [inline src/rewards/airdrop.ts:212]: Filtering is generally good
The datasize filter is welcome, it filters out all the old corrupted airdrops
I think you want the stake vault filter as well, we don't care about other stakepools

```suggestion
  static async loadAll(rewardsProgram: Program, stakePoolVault: PublicKey): Promise<Airdrop[]> {
    const dataSizeFilter: DataSizeFilter = {
      dataSize: 400464
    }
    const stakePoolFilter: MemcmpFilter = {
      memcmp: {
        offset: 8,
        bytes: stakePoolVault.toBase58()
      }
    }
    const airdropInfos = await rewardsProgram.account.airdrop.all([dataSizeFilter, stakePoolFilter])
```

**Review by Tristyn — COMMENTED** [review]
What do you think about keeping the stake pool filter

File: src/rewards/airdrop.ts (hunk 1 above)
Comment (qiqicodes) [reply src/rewards/airdrop.ts]: good idea =]

File: src/rewards/airdrop.ts [hunk 2]
```diff
@@ -206,14 +206,17 @@ export class Airdrop {
   }
 
   static async loadAll(rewardsProgram: Program, stakePoolVault: PublicKey): Promise<Airdrop[]> {
+    const dataSizeFilter: DataSizeFilter = {
+      dataSize: 400464
+    }
     const stakePoolFilter: MemcmpFilter = {
       memcmp: {
-        offset: 8,
+        offset: 112,
```
Comment (Tristyn) [inline src/rewards/airdrop.ts:224]:
```suggestion
        offset: 8 + 32 + 32 + 32 + 8, // airdop.stake_pool
```

**Review by Tristyn — APPROVED** [review]

**Review by Tristyn — APPROVED** [review]

**Pull request merged**